lib/puppet-validator.rb

```diff
@@ -73,19 +75,59 @@ def initialize(app=nil)
     erb :index
   end
 
+  get '/load/*' do
+    location = params[:splat].first.sub(/(https?:)\/\/?/, '\1//')
+    logger.info "Loading code from: #{location}"
+
+    uri = location.downcase.start_with?('http') ? URI.parse(location) : URI.parse("https://#{location}")
+
+    case uri.host
+    when 'gist.github.com'
+      code = open("#{uri}/raw").read
+    when 'pastebin.com', 'hastebin.com'
+      code = open("#{uri.scheme}://#{uri.host}/raw#{uri.path}").read
+    else
+      logger.info "Unrecognized paste service: #{uri}"
```

Comment:
Should the rest of the code in this endpoint run if the paste service is unrecognized?  What would `erb :index` contain in that case?

Reply:
The code box will be blank, but no other issues. I could make that more explicit.